### ISSUE TITLE
Remove default "main" cluster

### ISSUE THREAD

**Issue by nsagnett** [kubernetes-sigs/headlamp#2599]
I have an empty main cluster due to overriding default kubeconfig

## Describe the solution you'd like

I wonder to complete an in-cluster deployment with customs kubeconfig files without main cluster

## What users will benefit from this feature?

In-Cluster users (may be others also)

**Comment by joaquimrocha** [comment]
@nsagnett Thanks for reporting. We need more context. Can you tell us more about how your setup is done? (i.e. are you running with -in-cluster but also specifying a -kubeconfig and you do not want the cluster named `main` to be visible?)

**Comment by nsagnett** [comment]
Of course! So yes, indeed, it's in cluster setup and I override "default" kubeconfig, but even with that, "main" stay visible

**Comment by nsagnett** [comment]
The problem is this param is automatically added when I deploy the chart on cluster

**Comment by SegFault02** [comment]
it's hardcoded here https://github.com/headlamp-k8s/headlamp/blob/5d7870609f3630b5e579ab5ada771ebf0b1ba146/charts/headlamp/templates/deployment.yaml#L128

**Comment by illume** [comment]
cc @knrt10

**Comment by illume** [comment]
The -in-cluster is used internally for a number of features and config choices.

So, we can't just remove the -in-cluster param?

**Comment by lszczepaniak-XM** [comment]
hey, what is the progress of it? For my use case the default main cluster is not accessible through oidc, since it has wrong clusterURL, which for all clusters is a proxy to get the `id_token`. So in other words it is simply an error placeholder in the ui. Is there a way to edit the main cluster config or simply use any argument to stop loading it?
```
    {
      "name": "main",
      "server": "<incorrectURL, needs to be proxy URL>",
      "auth_type": "oidc",
      "meta_data": {
        "clusterID": "",
        "extensions": null,
        "namespace": "",
        "origin": {
          "kubeconfig": ""
        },
        "originalName": "main",
        "source": "incluster"
      }
    },
```
All other clusters follow this syntax:
```    {
      "name": "<actual cluster name>",
      "server": "<correct proxy URL>",
      "auth_type": "oidc",
      "meta_data": {
        "clusterID": "/home/headlamp/.config/Headlamp/kubeconfigs/config+<actual cluster name>",
        "extensions": {

        },
        "namespace": "",
        "origin": {
          "kubeconfig": "/home/headlamp/.config/Headlamp/kubeconfigs/config"
        },
        "originalName": "<actual cluster name>",
        "source": "dynamic_cluster"
      }
    },
```

**Comment by bbourchanin** [comment]
kubeconfig loaded htrough sidecar to /home/headlamp/.config/Headlamp/kubeconfigs/config works for myself (https://headlamp.dev/docs/latest/installation/in-cluster/#use-a-non-default-kube-config-file) - i add  this in the values to point my "default" cluster and make "main" disappear :
```
extraArgs:
              - "--in-cluster-context-name=mydefaultcluster"
```

**Comment by Pritz395** [comment]
@illume I can work on this issue right?
